tests/api/test_pagination.py

```diff
@@ -0,0 +1,1654 @@
+                "zordon@example.com",
+            ],
+        ),
+        (
+            {"field": "FIRST_NAME", "direction": "ASC"},
+            UserSortField.FIRST_NAME.value,
+            [
+                "zordon@example.com",
+                "leslie@example.com",
+                "ddevito@example.com",
+                "ellen@example.com",
+                "allen@example.com",
+            ],
+        ),
+        (
+            {"field": "LAST_NAME", "direction": "ASC"},
+            UserSortField.LAST_NAME.value,
+            [
+                "allen@example.com",
+                "ellen@example.com",
+                "ddevito@example.com",
+                "zordon@example.com",
+                "leslie@example.com",
+            ],
+        ),
+        (
+            {"field": "EMAIL", "direction": "ASC"},
+            UserSortField.EMAIL.value,
+            [
+                "allen@example.com",
+                "ddevito@example.com",
+                "ellen@example.com",
+                "leslie@example.com",
+                "zordon@example.com",
+            ],
+        ),
+        (
+            {"field": "EMAIL", "direction": "DESC"},
+            UserSortField.EMAIL.value,
+            [
+                "zordon@example.com",
+                "leslie@example.com",
+                "ellen@example.com",
+                "ddevito@example.com",
+                "allen@example.com",
+            ],
+        ),
+    ],
+)
+def test_user_pagination(
+    order_by,
+    cursor_fields,
+    user_order,
+    staff_api_client,
+    permission_manage_users,
+    users_for_pagination,
+):
+    query = """
+        query (
+            $first: Int, $last: Int, $after: String,
+            $before: String, $sortBy: UserSortingInput
+         ){
+          customers(
+            first: $first, last: $last, after: $after,
+            before: $before, sortBy: $sortBy
+        ) {
+            edges {
+              node {
+                firstName
+                lastName
+                email
+              }
+            }
+            pageInfo{
+              startCursor
+              endCursor
+              hasNextPage
+              hasPreviousPage
+            }
+          }
+        }
+    """
+    start_cursor_page1 = create_cursor_from_user_email(user_order[0], cursor_fields)
+    start_cursor_before = create_cursor_from_user_email(user_order[1], cursor_fields)
+    end_cursor_page1 = create_cursor_from_user_email(user_order[2], cursor_fields)
+    start_cursor_page2 = create_cursor_from_user_email(user_order[3], cursor_fields)
+    end_cursor_page2 = create_cursor_from_user_email(user_order[4], cursor_fields)
+
+    _test_pagination(
+        staff_api_client,
+        query,
+        order_by,
+        permission_manage_users,
+        start_cursor_page1,
+        end_cursor_page1,
+        start_cursor_page2,
+        end_cursor_page2,
+        start_cursor_before,
+        resolver_name="customers",
+    )
```

Comment:
It's impossible to tell what this test is checking as all the assertions are hidden inside a private helper.